tests/fixtures/parsers/typescript-parsers/array-callback-return.js

```diff
@@ -0,0 +1,320 @@
+"use strict";
+
+/**
+ * Parser: typescript-eslint-parser-3.0.0
+ * Source code:
+ * [0,1,2].map((num: number) => {
```

Comment:
I used code that produces lint errors when the associated lint rule is enabled. Is that the source to be expected when generating these test fixtures?

Can you share a bit about how these fixtures are structured and what they test exactly?
Thanks!